### ISSUE TITLE
Don't check Inv in state 0 with --init=Inv

### ISSUE THREAD

**Issue by nano-o** [informalsystems/apalache#1825]
Apalache checks that `Inv` holds in state 0 even with option `--init=Inv`. This is unnecessary but takes time, so I suggest skipping this check.

**Comment by nano-o** [comment]
Ran into a variant of this again today, where it takes about 10 minutes to check that `I /\ J => J`... Maybe what would be needed is to first apply some basic simplification rules to discharge obviously tautological goals.

**Comment by konnov** [comment]
Yeah, totally, it annoys me all the time too. The quick workaround is to pass:

**Comment by will62794** [comment]
+1. I also encountered this as a minor annoyance when checking a larger inductive invariant.

**Comment by konnov** [comment]
I have just lost half a day waiting for the invariant being checked against the initial states (totally forgot about that). Then the actual invariant violation was found in 1h. It's actually quite annoying for large specs. I am still struggling with finding a good UX for that though.

We had an idea of introducing profiles some time ago. For instance, this is how we could use a profile for an inductive step:

```sh
$ apalache-mc check --profile=inductive-step --init=IndInit --inv=IndInv spec.tla
```

In this profile, we could automatically:
 - enable invariant testing for state 1 only,
 - set `--length` to 1,
 - potentially, tune z3 parameters to something that better fits inductive invariants, see #2990.

**Comment by shalashaska117** [comment]
Opened #3395. It makes the common case automatic: when every `--inv` operator equals the `--init` operator (e.g. `--init=IndInv --inv=IndInv`), the state-0 invariant check is skipped, with a one-line log notice. Sound because the initial states satisfy the invariant by construction; violations after `Next` are still found (mayChangeAssertion forces a later-step check). I kept it to the smallest automatic behavior matching the issue title rather than the broader `--profile` idea - happy to gate it behind a flag or fold it into a profile if you prefer.

**Comment by konnov** [comment]
I think the main problem with issues like this one is that they are highly specific to a small subset of use cases. My current understanding is that we should keep the core of Apalache small and solve narrow use cases via other tools and scripts. For instance, checking inductive invariants can be done by writing a small script on top of JSON-RPC: https://github.com/apalache-mc/apalache/tree/main/json-rpc.
